src/policy/semantic.rs

```diff
@@ -187,6 +216,67 @@ impl<Pk: MiniscriptKey> Policy<Pk> {
         }
     }
 
+    fn _pprint_diff(&self, other: &Policy<Pk>, prefix: String, last: bool) {
+        match (self, other) {
+            (x, y) if x == y => x._pprint_tree(prefix, last),
+            (Policy::Threshold(k1, subs1), Policy::Threshold(k2, subs2))
+                if k1 == k2 && subs1.len() == subs2.len() =>
+            {
+                let prefix_current = if last { "`-- " } else { "|-- " };
+
+                println!("{}{}{}", prefix, prefix_current, self.term_name());
+
+                let prefix_child = if last { "    " } else { "|   " };
+                let prefix = prefix + prefix_child;
+
+                let last_child = subs1.len() - 1;
+
+                // Hashsets to maintain which children of subs are matched
+                // onto children from others
+                let mut ind_a = HashSet::new();
+                let mut ind_b = HashSet::new();
+                for i in 0..subs1.len() {
+                    let sub_a = &subs1[i];
+                    let b_pos = subs2.iter().position(|sub_b| sub_a == sub_b);
+                    match b_pos {
+                        Some(j) => {
+                            ind_a.insert(i);
+                            ind_b.insert(j);
+                        }
+                        None => {}
+                    }
+                }
+                let mut j = 0;
+                for (i, sub_a) in subs1.iter().enumerate() {
+                    // The position in subs2
+                    if ind_a.contains(&i) {
+                        sub_a._pprint_tree(prefix.to_string(), last);
+                    } else {
+                        while ind_b.contains(&j) {
+                            j = j + 1;
+                        }
+                        sub_a._pprint_diff(&subs2[j], prefix.to_string(), i == last_child);
+                    }
+                }
+            }
+            (x, y) => {
+                let red = "\x1b[0;31m";
+                let nc = "\x1b[0m";
+                let green = "\x1b[0;32m";
```

Comment:
I think we need to come up with a different encoding scheme. It doesn't have to be pretty, we could provide a shellscript or something that converted it into a colorized version.

But using shell escape sequences means that we won't work on windows and are likely to confuse other command line tools.